### ISSUE TITLE
New feature suggestion: Wiki Help Links

### ISSUE THREAD

**Issue by d3fz** [ericpaulbishop/gargoyle#669]
**Problem:**
New/existing users can get a little confused when it comes to getting that awesome feature working the way they want to (or working at all). Currently, the only way users seek for help is through the forums (apparently not everyone even realize there's a documentation available), which can take some time to get a response and more often than not, asking questions that have been answered before, over and over again.

**Proposal:**
I've been thinking in a better way to help users, perhaps directly through the UI. The idea is to add a link on specific features/sections (e.g. OpenVPN, QoS, other complicated ones), which basically will open a new window/tab that redirects users to the respective online wiki tutorial. That would save a lot of time (both users and admins) and point users in the right direction right away. If the user still has any questions, then they can always open a new thread on the forums.

Something like this:
![tgdhr8c](https://user-images.githubusercontent.com/22578839/31633219-7084d3c0-b29e-11e7-9f7c-a6d575c8307d.png)
![njyvfpw](https://user-images.githubusercontent.com/22578839/31633278-9a488e86-b29e-11e7-83eb-752e6eb5c816.png)

The idea has already been discussed a little on the forums (with @lantis1008 and @ispyisail), and a viable approach would be doing something similar to i18n, where: 

Example, using strings.js (currently read by all pages): 
`UI.HelpBridgeRepeater="www.gargoyle-router.com/wiki/doku.php?id=bridge_repeater";`
`UI.HelpMsg="Need help with this ?";`

`<a href='<%~ HelpBridgeRepeater %>'><%~ HelpMsg %></a>`

Or even redirect straight to a specific section:
`<a href='<%~ HelpQoS %>#active_congestion_controller_acc'><%~ HelpMsg %></a>`

@lantis1008 came up with this syntax:
`<span class="ext_help"><a href="<%~ external_help.ggweb %><%~ external_help.hlpBrdgcfg %>"><%~ external_help.HelpText %></a></span>`

Thoughts ?

**Comment by lantis1008** [comment]
My only comment is that my syntax is quite long. If we are concerned about bloating the SH files I can look at shrinking it. 
Also looking at options to turn off the help text for people who don’t want/need it. 
And of course, themes can hide them as well.

**Comment by obsy** [comment]
Not "need help with this", just question mark...

**Comment by lantis1008** [comment]
^ that would make the syntax much shorter on its own.

**Comment by ispyisail** [comment]
Netgear use "? Help"

![netgear](https://user-images.githubusercontent.com/12116110/31652519-e733c8ea-b37b-11e7-8152-84aed32cd9a7.jpg)

**Comment by d3fz** [comment]
Yes, I purposely used a very explicit message in this example, just to show what kind of action the user would look for/expect when clicking on the link. The message can always be shortened, which would go inline with the "More Info" links, found on multiple pages. 

I understand that using only a question mark character would clutter less, my only concern is that it simply goes unnoticed to the user, unless it's done in a way that draws more attention to it. GitHub's approach looks nice and simple, using icon + tooltip when hovered:
 
![laivjgc - imgur](https://user-images.githubusercontent.com/22578839/31837360-444fc6a4-b5b7-11e7-8fa4-aa48782e6bbb.png)

Also, if the syntax length is a problem, why don't just add new lines to strings.js and pull from there ? Plus the advantage of not needing to create an additional file, as all missing strings (on another language) would fallback to the default strings.js (EN). 

Example:
```
UI.GWeb="www.gargoyle-router.com/wiki/doku.php?id=";
UI.HlpBasic="basic";
UI.HlpQuotas="quotas";
UI.HlpQoS="qos";
UI.HlpOpenVPN="openvpn";
UI.HlpBridgeRepeater="bridge_repeater";
UI.HlpTxt="Need help?";
```
Where would wrap up like this:
```
<span class="help">
    <a href="<%~ GWeb %><%~ HlpQoS %>"><%~ HlpTxt %></a>
</span>
```
Thoughts ?

**Comment by lantis1008** [comment]
I’ve already done something similar, but it still adds lots of characters to the SH files which won’t be compressed. 
I just remembered that the language JS aren’t compressed either

**Comment by lantis1008** [comment]
Gargoyle already has code to generate pop up text exactly like that GitHub example :)

**Comment by ispyisail** [comment]
The other suggestion is my netgear router just brings up a text box

![2017-10-21 21_37_39-netgear prosafe - router status](https://user-images.githubusercontent.com/12116110/31850024-71502fe4-b6a8-11e7-8d57-8d956e6881c1.jpg)

Dis-advantage

* More work for English speakers
* No pictures or diagrams
* More router space required?

Advantage 

* Easy to use Google translate for more languages
* Links never good down in case of internet loss
* You could still put wiki links in the text box for further clarification 

I'm not sure its a better idea but I'm just putting it out there
